File: BOJ/#1065 한수/한수.cpp
#include <iostream>
using namespace std;

bool isProg(int n) {
    if(n < 100) return true;

    int last = n%10;
    n /= 10;
    int sub = last-(n%10);

    while(n >= 10) {
        last = n%10;
        n /= 10;

        if(last-n%10 != sub) return false;
    }

    return true;
}

int main() {
    int N; cin>>N;

    int cnt = 0;
    for(int i=1; i<=N; i++)
        if(isProg(i)) cnt++;

    cout<<cnt;
}

File: BOJ/#1065 한수/한수.py
def isProg(n):
    if n < 100: return True

    last = n%10
    n //= 10
    sub = last-n%10

    while n >= 10:
        last = n%10
        n //= 10

        if last-n%10 != sub: return False

    return True

def main():
    N = int(input())

    cnt = 0
    for i in range(1, N+1):
        if isProg(i): cnt += 1

    print(cnt)

main()
